### PR TITLE
Non-public cloud support for Azure Key Vault.

### LINKED ISSUE #2733: Support non-public clouds for Azure Key Vault authentication
### Proposal

Support non-public clouds for Azure Key Vault authentication

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by v-shenoy**
Signed-off-by: Vighnesh Shenoy <vshenoy@microsoft.com>

Azure Key Vault now supports non-public clouds via the `cloud` parameter.

Sample TriggerAuthentication definition with keyvault.
```yaml
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: triggerauth-name
  namespace: default
spec:
  azureKeyVault:
    vaultUri: <vault-uri>
    credentials:  
      clientId: <active-directory-client-id>
      clientSecret:
        valueFrom:
          secretKeyRef:
            name: <secret-containing-client-password>
            key: <key-within-secret>
      tenantId: <active-directory-tenant-id>
    cloud: # Optional
      type: private # Required
      keyVaultResourceURL: <key-vault-resource-url-for-cloud> # Required when type = private
      activeDirectoryEndpoint: <active-directory-endpoint-for-cloud> # Required when type = private
    secrets: 
    - parameter: <param-name-for-scaled-object>
      name: <secret-name-in-key-vault>
```

Documentation PR - https://github.com/kedacore/keda-docs/pull/732

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Relates to #2733

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Might be good to align the cloud information with scalers maybe?

What about this?
```diff
-cloudInfo: # Optional
+cloud:
  type: private # Required
-  keyVaultResourceURL: <key-vault-resource-url-for-cloud> # Required when type = private
+  keyVaultResourceUri: <key-vault-resource-url-for-cloud> # Required when type = private
-  activeDirectoryEndpoint: <active-directory-endpoint-for-cloud> # Required when type = private
+  activeDirectoryUri: <active-directory-endpoint-for-cloud> # Required when type = private
```

**Comment by tomkerkhove** [comment]
Can you open a PR for our docs and link it in this PR please? This is why I tend to use "relates to" instead of "fixes" as we always need a link without closing on this PR.

**Comment by v-shenoy** [comment]
> Can you open a PR for our docs and link it in this PR please? This is why I tend to use "relates to" instead of "fixes" as we always need a link without closing on this PR.

Was away for some time, am writing the docs as we speak. I don't get the second part of your comment, can you elaborate a bit?

**Comment by v-shenoy** [comment]
Doc PR - https://github.com/kedacore/keda-docs/pull/732

**Comment by v-shenoy** [comment]
> 

Didn't change the `keyVaultResourceURL` and `activeDirectoryEndpoint` as the names align with the changes done for EventHub and Application Insights.

**Comment by tomkerkhove** [comment]
Fine by me!

**Comment by tomkerkhove** [comment]
> > Can you open a PR for our docs and link it in this PR please? This is why I tend to use "relates to" instead of "fixes" as we always need a link without closing on this PR.
> 
> Was away for some time, am writing the docs as we speak. I don't get the second part of your comment, can you elaborate a bit?

Neermind the second part!

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@
 - **General:** Introduce new GCP Storage Scaler ([#2628](https://github.com/kedacore/keda/issues/2628))
 - **General:** Introduce ARM-based container image for KEDA ([#2263](https://github.com/kedacore/keda/issues/2263)|[#2262](https://github.com/kedacore/keda/issues/2262))
 - **General:** Provide support for authentication via Azure Key Vault ([#900](https://github.com/kedacore/keda/issues/900))
+- **General:** Provide support for non-public clouds for Azure Key Vault ([#2733](https://github.com/kedacore/keda/issues/2733))
```
Comment (zroubalik) [inline CHANGELOG.md:39]: Just a nit: I'd merge this with the previous entry and just add the second link to the issue 2733 (the same as it is done on line 37).

Comment (v-shenoy) [reply CHANGELOG.md]: Done.

**Review by zroubalik — APPROVED** [review]
LGTM

**Review by JorTurFer — APPROVED** [review]
LGTM

**Comment by JorTurFer** [comment]
Thanks for this improvement! ❤️

**Pull request merged**